src/devices/cddadevice.cpp

```diff
@@ -73,12 +71,28 @@ void CddaDevice::WatchForDiscChanges(bool watch) {
     disc_changed_timer_.stop();
 }
 
-void CddaDevice::LoadSongs() { cdda_song_loader_.LoadSongs(); }
+void CddaDevice::ForceLoadSongs() {
+  cdda_song_loader_.LoadSongs();
+  disc_changed_timer_.stop();
+}
+
+void CddaDevice::LoadSongs() {
+  SongList songs = cdda_song_loader_.chached_tracks();
+  SongsLoaded(songs);
+}
```

Comment:
Maybe there's a better name for this method. It isn't actually loading the tracks anymore.

Reply:
True. Any suggestions? Maybe `RequestSongs` or `AnnounceSongs`/`SignalSongs`?

Reply:
Hmm. Maybe `UpdateFromLoader`?

Reply:
But it's also not really updating anything, using just the cached information. The only thing it really does is (re)-emit the signal with the known song list (which gets updated automatically upon media change or on first read). So maybe just `EmitSongs`?

Thinking about it, it seems to indicate that using the signal as the main way of getting the song list to the consumers might not be the best design decision, because it forces this odd re-emission. I could change it so that the `CddaDevice` (and potentially also `CddaSongLoader`) just expose methods to get the song information simply emit a signal whenver that changes. Consumers can then just call the method when receiving the signal. That would also take care of the concern you have expressed in the discussion below, as the signal would not be re-emitted if a new consumer wanted to query the tracks.

Reply:
Yeah, as long as the fetch is re-entrant, which I think it is.

Reply:
I'm not quite sure it is. I have now done the following:
- protected access to the tracklist cache in CddaSongLoader with a mutex
- added method CddaDevice::songs (which forwards to CddaSongLoader::cached_tracks)
- removed CddaDevice::LoadSongs
- renamed the private CddaDevice::ForceLoadSongs back to CddaDevice::LoadSongs
- RipCDDialog now invokes CddaDevice::songs to get the song list from the device instead of provoking it to emit a signal

I didn't change the behavior of the existing signals, so those still provide the updated list of songs to the receiver.